Source/carthage/Build.swift

```diff
@@ -163,34 +163,43 @@ public struct BuildCommand: CommandProtocol {
 		}
 	}
 
-	/// Opens a temporary file on disk, returning a handle and the URL to the
+	/// Opens an existing file, if provided, or creates a temporary file if not, returning a handle and the URL to the
 	/// file.
-	private func openLogFile(_ path: String?) -> SignalProducer<(FileHandle, URL), NSError> {
+	private func openLogFile(_ path: String?) -> SignalProducer<(FileHandle, URL), CarthageError> {
 		return SignalProducer.attempt {
-			let logFD: Int32
-			let logPath: String
-			
 			if let path = path {
-				logFD = FileHandle(forUpdatingAtPath: path)?.fileDescriptor ?? -1
-				logPath = path
+				if !FileManager.default.fileExists(atPath: path) {
+					FileManager.default.createFile(atPath: path, contents: nil, attributes: nil)
+				}
+				
+				let fileURL = URL(fileURLWithPath: path, isDirectory: false)
+				
+				guard let handle = FileHandle(forUpdatingAtPath: path) else {
```

Comment:
And actually, the original bug shows that that's the case. (Because the `FileHandle` wasn't retained, and the file descriptor was returned, but was no longer valid because the `FileHandle` had deallocated.)